.github/workflows/reusable-docker.yml

```diff
@@ -0,0 +1,141 @@
+name: Build, tag and push docker image to ghcr.io
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+on:
+  workflow_call:
+    inputs:
+      environment:
+        required: true
+        type: string
+      package_name:
+        required: true
+        type: string
+      project_path:
+        required: true
+        type: string
+      depot_project_id:
+        required: true
+        type: string
+      test_port:
+        required: false
+        default: '1341'
+        type: string
+      health_check:
+        required: false
+        default: false
+        type: boolean
+      # local tag is used to tag the image locally, you can find it in package.json
+      local_tag:
+        required: true
+        type: string
+      # supported tags: dev, stg, prod
+      env_tag:
+        required: false
+        type: string
+    outputs:
+      docker_image:
+        description: 'The image that was built'
+        value: ${{ jobs.reusable_docker.outputs.docker_image }}
+      docker_image_ee:
+        description: 'The enterprise image that was built'
+        value: ${{ jobs.reusable_docker.outputs.docker_image_ee }}
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  reusable_docker:
+    runs-on: ubuntu-latest
+    timeout-minutes: 80
+    environment: ${{ inputs.environment }}
+    outputs:
+      docker_image: ${{ steps.save-image-to-output.outputs.IMAGE }}
+      docker_image_ee: ${{ steps.save-image-to-output.outputs.IMAGE_EE }}
+    permissions:
+      contents: read
+      packages: write
+      deployments: write
+      id-token: write
+    strategy:
+      matrix:
+        name: [ '${{ inputs.package_name }}-ee', '${{ inputs.package_name }}' ]
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: ./.github/actions/setup-project
+        with:
+          slim: 'true'
+
+      - name: Setup Depot
+        uses: depot/setup-action@v1
+        with:
+          oidc: true
+
+      - name: Build, tag, and push image to ghcr.io
+        id: build-image
+        if: ${{ inputs.env_tag == 'dev' || inputs.env_tag == 'stg' }}
+        env:
+          REGISTRY_OWNER: novuhq
+          DOCKER_NAME: ${{ matrix.name }}
+          LOCAL_TAG: ${{ inputs.local_tag }}
+          IMAGE_TAG: ${{ github.sha }}
+          ENV_TAG: ${{ inputs.env_tag }}
+          PROJECT_PATH: ${{ inputs.project_path }}
+          GH_ACTOR: ${{ github.actor }}
+          GH_PASSWORD: ${{ secrets.GH_PACKAGES }}
+          DEPOT_PROJECT_ID: ${{ inputs.depot_project_id }}
+        run: |
+          echo $GH_PASSWORD | docker login ghcr.io -u $GH_ACTOR --password-stdin
+          cd $PROJECT_PATH && npm run docker:build:depot
```

Comment:
it uses the `docker:build:depot` command that should be present in the `package.json` of the build project